specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json

```diff
@@ -0,0 +1,1956 @@
+                    "name": "AllocationState",
+                    "modelAsString": true
+                  }
+                },
+                "allocationStateTransitionTime": {
+                  "type": "string",
+                  "readOnly": true,
+                  "format": "date-time",
+                  "title": "Allocation state transition time.",
+                  "description": "The time at which the compute entered its current allocation state."
+                },
+                "errors": {
+                  "readOnly": true,
+                  "title": "Errors.",
+                  "description": "Collection of errors encountered by various compute nodes during node setup.",
+                  "type": "array",
+                  "items": {
+                      "$ref": "#/definitions/MachineLearningServiceError"
+                  }
+                },
+                "currentNodeCount": {
+                  "type": "integer",
+                  "readOnly": true,
+                  "format": "int32",
+                  "title": "Current node count.",
+                  "description": "The number of compute nodes currently assigned to the compute."
+                },
+                "targetNodeCount": {
+                  "type": "integer",
+                  "readOnly": true,
+                  "format": "int32",
+                  "title": "Target node count.",
+                  "description": "The target number of compute nodes for the compute. If the allocationState is resizing, this property denotes the target node count for the ongoing resize operation. If the allocationState is steady, this property denotes the target node count for the previous resize operation."
+                },
+                "nodeStateCounts": {
+                  "title": "Node state counts.",
+                  "description": "Counts of various node states on the compute.",
+                  "readOnly": true,
+                  "$ref": "#/definitions/NodeStateCounts"
+                }
+              }
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "AmlCompute"
+    },
+    "VirtualMachine": {
+      "description": "A Machine Learning compute based on Azure Virtual Machines.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Compute"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "type": "object",
+              "properties": {
+                "virtualMachineSize": {
+                  "description": "Virtual Machine size",
+                  "type": "string"
+                },
+                "sshPort": {
+                  "description": "Port open for ssh connections.",
+                  "type": "integer"
+                },
+                "address": {
+                  "description": "Public IP address of the virtual machine.",
+                  "type": "string"
+                },

```

Comment:
what is the format of this ip address? ipv4 or fqdn?

Reply:
we accept both